sources/multi.lanraragi/src/lib.rs

```diff
@@ -0,0 +1,310 @@
+#![no_std]
+extern crate alloc;
+
+mod models;
+mod settings;
+
+use crate::models::{Archive, ArchiveMetadata, Category, SearchResult};
+use aidoku::{
+	BaseUrlProvider, Chapter, DynamicFilters, Filter, FilterValue, Manga, MangaPageResult,
+	MangaStatus, Page, PageContent, Result, SelectFilter, Source,
+	alloc::{String, Vec},
+	helpers::uri::encode_uri,
+	imports::net::Request,
+	prelude::*,
+};
+use alloc::string::ToString;
+use alloc::vec;
+
+fn extract_date_added(tags: &str) -> Option<i64> {
+	tags.split(',').find_map(|tag| {
+		let tag = tag.trim();
+		if tag.starts_with("date_added:") {
+			tag.strip_prefix("date_added:")?.parse::<i64>().ok()
+		} else {
+			None
+		}
+	})
+}
+
+static mut CURRENT_START: i32 = 0;
+
+struct Lanraragi;
+
+impl Source for Lanraragi {
+	fn new() -> Self {
+		Self
+	}
```

Comment:
you shouldn't use static muts if possible. you can add a field to the source struct to use instead.

```suggestion
#[derive(Default)]
struct Lanraragi {
	current_start: RefCell<i32>,
}

impl Source for Lanraragi {
	fn new() -> Self {
		Self::default()
	}
```